### PR TITLE
RSDK-3262 Base station RTK improvements

### PR DESCRIPTION
**Pull request by oliviamiller**
### Summary
This PR includes many improvements to rtk base station component. There are breaking changes to the gpsrtk and rtk base station. I didn't include any NMEA gps breaks in this PR but documented some changes that could be made [here](https://viam.atlassian.net/browse/RSDK-3525). Changes in this PR: 

- fixes bug that errors when trying to read configs 
- removes ntrip option as a correction source. A base station can send correction data through serial or i2c
- removes surveyIn config value since no longer needed with no ntrip option
-  moves board to the I2C attributes 
- The children of a base station now should be RTK gps models, not NMEA models
- clean up unit tests in rtk_station_tests

### Testing
Tested manually with serial and i2c.

### CODE REVIEW

File: components/movementsensor/gpsrtk/ntrip.go
Comment (susmitaSanyal) [inline components/movementsensor/gpsrtk/ntrip.go:30]: we should reorganize this while we are at it. Let's move all the RTK station stuff out of gpsrtk. Make gpsrtk look a lot cleaner.

File: components/movementsensor/gpsrtk/gpsrtk.go
Comment (susmitaSanyal) [inline components/movementsensor/gpsrtk/gpsrtk.go:69]: if it's only for testing, should we move this to test? Can't we use a fake path for testing? Maybe I'm not understanding the reason for this Test channel.

File: components/movementsensor/gpsrtk/ntrip.go
Comment (susmitaSanyal) [inline components/movementsensor/gpsrtk/ntrip.go:286]: Nice job making ntrip readable!

File: components/movementsensor/gpsrtk/rtk_station.go
Comment (susmitaSanyal) [inline components/movementsensor/gpsrtk/rtk_station.go:50]: required Accuracy can be 1-5 <- adding unit here would be nice.

**Review by susmitaSanyal — CHANGES_REQUESTED** [review]

File: components/movementsensor/gpsrtk/rtk_station.go
Comment (oliviamiller) [reply components/movementsensor/gpsrtk/rtk_station.go:50]: I don't think there is a unit it's just an arbitrary ranking, but I can add in that 5 is the highest accuracy.

Comment (susmitaSanyal) [reply components/movementsensor/gpsrtk/rtk_station.go:50]:
```suggestion
	errRequiredAccuracy  = errors.New("required Accuracy can be a fixed number in the range of 1-5, 5 being the highest accuracy")
```

Comment (randhid) [reply components/movementsensor/gpsrtk/rtk_station.go:50]: I don't actually think this is true.

Comment (randhid) [reply components/movementsensor/gpsrtk/rtk_station.go:50]: I vaguely remember a fix of 4 being best, but I can't find a resource on this. Happy to keep the error as is until I find the resource confirming this

File: components/movementsensor/gpsrtk/ntrip.go
Comment (oliviamiller) [reply components/movementsensor/gpsrtk/ntrip.go:30]: Can you be more specific about what you mean here? I think the rtk_station is already seperate from gpsrtk.

File: components/movementsensor/gpsrtk/gpsrtk.go
Comment (randhid) [reply components/movementsensor/gpsrtk/gpsrtk.go:69]: This doesn't render on the app, by the way, it does not create a json config field there.

File: components/movementsensor/gpsrtk/ntrip.go
Comment (oliviamiller) [reply components/movementsensor/gpsrtk/ntrip.go:30]: Oh yeah definitely might make sense to move them to separate packages, I think that can be addressed in a separate PR once the scope doc is approved since we are likely changing base_station to the sensor api and moving to viam labs.

Comment (randhid) [reply components/movementsensor/gpsrtk/ntrip.go:30]: It is okay to change them before the scope is approved as long as the model names and config are retained and the drivers do the same thing as they do now. But happy to wait or do that in a separate PR. The work for moving them is nearly equivalent.

File: components/movementsensor/rtk_station/rtk_station.go
Comment (randhid) [inline components/movementsensor/rtk_station/rtk_station.go:222]: actually, do we need thhe rtkgps children anymore? since it's basically just writing to a serial port or an i2c address?

Comment (oliviamiller) [reply components/movementsensor/rtk_station/rtk_station.go:222]: Hm yeah..the only reason I could see it being needed is if some rovers are reading corrections from serial and some are reading from i2c...would that be possible?

Comment (randhid) [reply components/movementsensor/rtk_station/rtk_station.go:222]: Okay to do it in a eparate pr.

**Review by susmitaSanyal — APPROVED** [review]
LGTM

**Review by randhid — APPROVED** [review]
LGTM, let's start with the next bit of this project!

**Pull request merged**
